vkubelet/api/pods.go

```diff
@@ -0,0 +1,53 @@
+package api
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/cpuguy83/strongerrors"
+	"github.com/virtual-kubelet/virtual-kubelet/log"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+)
+
+// ContainerLogsBackend is used in place of backend implementations for the provider's pods
+type RunningPodsBackend interface {
+	// GetPods retrieves a list of all pods running on the provider (can be cached).
+	GetPods(context.Context) ([]*v1.Pod, error)
+}
+
+func RunningPodsHandlerFunc(p RunningPodsBackend) http.HandlerFunc {
+	scheme := runtime.NewScheme()
+	v1.SchemeBuilder.AddToScheme(scheme)
+	codecs := serializer.NewCodecFactory(scheme)
+
+	return handleError(func(w http.ResponseWriter, req *http.Request) error {
+		ctx := req.Context()
+		ctx = log.WithLogger(ctx, log.L)
+		pods, err := p.GetPods(ctx)
+		if err != nil {
+			return strongerrors.System(err)
```

Comment:
@cpuguy83 No idea if this is the right one to use.

Reply:
The provider should give the correct error rather than try to figure it out here.

Reply:
Will handlError handle it correctly if they don't return a strongerror?